oap-server/server-core/src/main/java/org/apache/skywalking/oap/server/core/cache/ServiceInstanceInventoryCache.java

```diff
@@ -119,4 +125,24 @@ public int getServiceInstanceId(int serviceId, int addressId) {
         }
         return serviceInstanceId;
     }
+
+    public String getServiceInstanceLanguage(int serviceInstanceId) {
+        String language = languageCache.getIfPresent(serviceInstanceId);
```

Comment:
I think comparing to this implementation, I would say, adding a `no-@column` field in the `ServiceInstanceInventory` to cache the status should be better in performance and understanding.

And you should avoid check the language repeatedly if this instance hasn't language properties. Such as in the new implementation, you should separate the value from `null` and `Const#UNKNOWN`.

Reply:
done